xlators/features/trash/src/trash.c

```diff
@@ -2246,6 +2246,9 @@ reconfigure(xlator_t *this, dict_t *options)
     GF_OPTION_RECONF("trash", priv->state, options, bool, out);
 
     if (priv->state) {
+        if (!priv->trash_itable)
+            priv->trash_itable = inode_table_new(0, this);
```

Comment:
Shouldn't we destroy the table if the option is disabled ? Otherwise we'll still waste memory, which I think it's what is trying to do this patch.

Has any side consequences destroying the table online ? if so, probably we shouldn't allow online modification of this option.

Reply:
Resolve the comment and push a patch with a new pull request(https://github.com/gluster/glusterfs/pull/1599) so deleting this pull reauest.